### ISSUE TITLE
List: improve splitting and combining lists

### ISSUE THREAD

**Issue by RolfKyburz** [WordPress/gutenberg#46039]
Splitting a list into two (e.g., to insert a paragraph between two list items), as well as combining two lists into one is clumsy and counter-intuitive. It actually always has been counter-intuitive, but I think it got worse in WP 6.1.1:

- I need to place the insert pointer to the end of the preceding list item, then type <return> — inserts a new list item, then I need to type <return> _again_ to get a paragraph block.

- Undoing this action is worse: using Backspace turns the paragraph into an empty list item in the list above. Typing Backspace again removes that empty list item, but still leaves me with two separate lists. Only typing forward-delete ultimately combines the lists fragments into one list.

### Environment info
WP 6.1.1, PHP 8.0.25, Theme: Exhibit 1.04, parent theme GeneratePress 3.2.3

**Comment by ndiego** [comment]
This issue was reviewed in today's Editor Bug Scrub and has been confirmed. Personally, I feel the second bullet point is the more confusing one. It would be great if the two separate list items were automatically merged into one. But I am not sure if this would mess people up who are actually trying to create two separate lists. 🤔

**Comment by sarahcada** [comment]
I was able to replicate this as well. The current behavior described above feels logical and intuitive to me. The first bullet point is consistent with the behaviors in other word processors like Google Docs. That said, I do I wonder, for folks that feel that this a clumsy experience, what action would be intuitive when one wants to insert a paragraph between two list items?

For the second bullet point, if one were to want two lists without anything between them, leaving an empty paragraph block between two lists would work. That empty paragraph won't be visible outside the editor.

**Comment by ellatrix** [comment]
> * I need to place the insert pointer to the end of the preceding list item, then type  — inserts a new list item, then I need to type  _again_ to get a paragraph block.

With "type", do you mean pressing Enter? What are you expecting? We can't immediately split a list, because how else do you insert a new list item?

I also think pressing backspace is quite intuitive. Here are the steps:

1. I start with two list items, place the caret at the end of the first one.

```
* one‸
* two
```

2. Press Enter to create new list item.

```
* one
* ‸
* two
```

3. If what I wanted was to split the list, I can press Enter again.

```
* one
‸
* two
```

4. Now if I want to undo it by pressing Backspace, it can be done again in two steps. First merge the paragraph into the list.

```
* one
* ‸
* two
```

5. Then press Backspace again to end up in the same spot.

```
* one‸
* two
```

Now you are correct that it doesn't combine the list, and that should probably be fixed. I'm not sure if there are more issues that I'm not understanding?

**Comment by RolfKyburz** [comment]
Thanks for looking into this. Yes, "type" meant "type Enter". Visually, i.e., at the frontend, the result is OK, but on the backend, the situation is annoying, e.g., in that "Select parent block (List)" only selects a part of the list. In general, handling lists is tricky and fragile (it's easy to end up with an invalid block). One example: creating an ordered list with non-consecutive numbers (& altering some other list properties) is hardly possible for people without coding knowledge.

**Comment by ellatrix** [comment]
> One example: creating an ordered list with non-consecutive numbers (& altering some other list properties) is hardly possible for people without coding knowledge.

Could you create a separate issue for this?

I have fixed this issue with #52995, awaiting review. If there's remaining issues, I think it would be best to file separate issues for those as well. Thank you!

**Comment by RolfKyburz** [comment]
Thanks — will do over the next couple of days. Will take some time to re-evaluate & document...
